pkg/utils/kubernetes/featuregates.go

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+import (
+	"fmt"
+
+	"github.com/gardener/gardener/pkg/utils/version"
+)
+
+// featureGateVersionRanges contains the version ranges for all Kubernetes feature gates.
+// Extracted from https://raw.githubusercontent.com/kubernetes/kubernetes/release-${version}/pkg/features/kube_features.go.
+// To maintain this list for each new Kubernetes version:
+// * Run hack/compare-k8s-feature-gates.sh <old-version> <new-version>.
+//   It will present 2 lists of feature gates: those added and those removed in <new-version> compared to <old-version>.
+// * Add all added feature gates to the map with <new-version> as MinVersion and no MaxVersion.
+// * For any removed feature gates, add <new-version> as MaxVersion to the already existing feature gate in the map.
```

Comment:
I think, setting `MaxVersion` to the version, in which a feature gate is removed, is confusing.
Open to other opinions, though.

Reply:
I did it like this for the following reasons:
* When comparing versions, I do `<` with MaxVersion right now. If MaxVersion is one version below, I wouldn't be able to do a direct comparison (`<=` or any other comparison operator doesn't work), so I would have to calculate the "+1" version (e.g. "1.16" from "1.15") and do `<` again. I thought it's not worth the trouble having in mind also the below 2 considerations.
* When maintaining versions, I find it less error prone to add the same version as `MinVersion` for added feature gates and `MaxVersion` for removed feature gates. With 2 different versions it would be easier to make mistakes (at least for me).
* As a Go (formerly C, C++, Java ...) developer I find intervals of type [a, b) quite familiar. They are used often, e.g. with the slice operator (':') in Go.

Reply:
Ok, thanks for explaining. You can keep it the way it is then.

Reply:
It comes to my mind, if you still find it confusing, I could change the naming to `AddedInVersion` / `RemovedInVersion` or something similar (forgoing the generic `version.Range` in favor of a featuregates-specifc type), while preserving the logic as it is. What do you think?

Reply:
That sounds good. I think `AddedInVersion` / `RemovedInVersion` is less confusing 👍

Reply:
@stoyanr While you have presented good arguments for keeping it, personally, I'd also be more in favor of `AddedInVersion` and `RemovedInVersion` - it's more meaningful and self-explaining (see also a comment from myself in an earlier review (https://github.com/gardener/gardener/pull/4149#discussion_r643673347)).

Reply:
Done